packages/svelte/src/compiler/phases/3-transform/client/visitors/javascript-runes.js

```diff
@@ -207,33 +207,23 @@ export const javascript_visitors_runes = {
 
 						seen.push(name);
 
-						let id = property.value;
-						let initial = undefined;
-
-						if (property.value.type === 'AssignmentPattern') {
-							id = property.value.left;
-							initial = /** @type {import('estree').Expression} */ (visit(property.value.right));
-						}
-
+						let id =
+							property.value.type === 'AssignmentPattern' ? property.value.left : property.value;
 						assert.equal(id.type, 'Identifier');
-
 						const binding = /** @type {import('#compiler').Binding} */ (state.scope.get(id.name));
+						let initial = /** @type {import('estree').Expression | null} */ (binding.initial);
+						if (initial) {
+							initial = /** @type {import('estree').Expression} */ (visit(initial));
+						}
```

Comment:
minor style point but how about this? one less type cast, one less unreassigned `let`

```suggestion
						const initial =
							binding.initial &&
							/** @type {import('estree').Expression} */ (visit(binding.initial));
```

Reply:
For some reason github marked this as outdated, will apply locally